src/App.tsx

```diff
@@ -1,8 +1,102 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import { PostList } from './components/PostList/PostList';
 
-const App = () => (
-  <h1>Dynamic list of posts</h1>
-);
+import {
+  getUsers,
+  getPosts,
+  getComments,
+  User,
+  Post,
+  Comment } from './helpers/api';
 
-export default App;
+export const App = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [comments, setCommnets] = useState<Comment[]>([]);
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filteredPosts, setFilteredPosts] = useState<Post[]>([]);
+  const [filterQuery, setFilterQuery] = useState('');
+
+  const handleClick = async () => {
+    setIsLoading(true);
+
+    try {
+      const postsFromServer = await getPosts();
+      const usersFromServer = await getUsers();
+      const commentsFromServer = await getComments();
+
+      setPosts(postsFromServer);
+      setFilteredPosts(postsFromServer);
+      setUsers(usersFromServer);
+      setCommnets(commentsFromServer);
+      setIsLoaded(true);
+      setErrorMessage('');
+    } catch (exeption) {
+      setErrorMessage('Error');
+    }
+
+    setIsLoading(false);
+  }
+
+  const changeFilterInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setFilterQuery(event.target.value);
+
+    setTimeout(() => {
+      setFilteredPosts(posts.filter(({ title, body }) => (title + body)
```

Comment:
store the initial array in the state, the one you wanna show in the list can be stored in the variable, which will be calculated right before the render